### DISCUSSION TITLE
Is there any ways/tools to get memory consumptions by dag/task?

### DISCUSSION THREAD

**Discussion by maksiplus19** [apache/airflow#32794] [Q&A] [answered]
We have two applications that share airflow. One have 7 constant dags, other build dags dynamicly. Some users complaint about workers use all memory and restart with OOM. How can we analyse memory usage?
Also can we get memory usage by dag parser?
Now we cant understand, is it normal memory usage and we need increase limits or some dags use too much memory

**Comment by potiuk** [comment, marked as answer]
Just the normal memory monitoring tools for python apps. Airflow is no different than any other python app.


I've heard good stories of memray https://github.com/bloomberg/memray
